packages/typegpu/src/tgpuBindGroupLayout.ts

```diff
@@ -528,28 +569,10 @@ class TgpuBindGroupImpl<
             };
           }
 
-          if ('externalTexture' in entry) {
-            let resource: GPUExternalTexture;
-
-            if (isExternalTexture(value)) {
-              resource = unwrapper.unwrap(value);
-            } else {
-              resource = value as GPUExternalTexture;
-            }
-
-            return {
-              binding: idx,
-              resource,
-            };
-          }
-
-          if ('sampler' in entry) {
+          if ('externalTexture' in entry || 'sampler' in entry) {
```

Comment:
why don't we separate it to return GPUExternalTexture for externalTexture and GPUSampler for sampler?

Reply:
Its the same JS behavior in both branches. We will probably split them later down the road when we refactor typed samplers.